modules/core/04-channel/keeper/upgrade.go

```diff
@@ -730,25 +743,25 @@ func (k Keeper) startFlushUpgradeHandshake(
 		return errorsmod.Wrapf(connectiontypes.ErrInvalidConnectionState, "connection state is not OPEN (got %s)", connectiontypes.State(connection.GetState()).String())
 	}
 
-	// the current upgrade handshake must only continue if both channels are using the same upgrade sequence,
-	// otherwise an error receipt must be written so that the upgrade handshake may be attempted again with synchronized sequences
-	if counterpartyChannel.UpgradeSequence != channel.UpgradeSequence {
-		// save the previous upgrade sequence for the error message
-		prevUpgradeSequence := channel.UpgradeSequence
+	return nil
+}
+
+// checkForUpgradeSequences ensures current upgrade handshake only continues if both channels are using the same upgrade sequence,
+// otherwise an upgrade error is returned so that an error receipt will be written so that the upgrade handshake may be attempted again with synchronized sequences.
+func (k Keeper) checkForUpgradeSequences(ctx sdk.Context, portID, channelID string, channel types.Channel, counterpartyUpgradeSequence uint64) error {
```

Comment:
I think the word `check` in the function name squeaks a bit for me: I would expect that the function perform only checks and does not mutate state (which does by updating the channel's upgrade sequence). Would you consider `syncUpgradeSequence` or `updateUpgradeSequence` instead?

Reply:
I like `syncUpgradeSequence`!